### DISCUSSION TITLE
Use ETH instead of STRK as gas

### DISCUSSION THREAD

**Discussion by azamtim** [software-mansion/starknet.py#1450] [General]
Hi all
I'm using folowing function to transfer tokens from one wallet to another:
 `  invocation = contract.functions["transfer"].invoke_v3_sync(
                    recipient=parse_address(destination_wallet), amount=balance, auto_estimate=True
                )`
                
However i get the message:
FAIL: Client failed with code 55. Message: Account validation failed. Data: perform_validations call failed; failure reason: TransactionFailureReason(code='native_blockifier.PyTransactionPreValidationError', error_message='TransactionFeeError(L1GasBoundsExceedBalance { max_amount: 29, max_price: 46018155035322, balance_low: StarkFelt("0x0000000000000000000000000000000000000000000000000000000000000000"), balance_high: StarkFelt("0x0000000000000000000000000000000000000000000000000000000000000000") })').

As I found out it tries to use STRK as gas, which parameter should i use in the function to make it using ETH as a gas

thanks

**Comment by ddoktorski** [comment]
Hi @azamtim
For transaction fees in ETH, you should use `invoke_v1`. Please note that all v3 transactions are using STRK as their fee token.

Let us know if you have any other questions

**Comment by azamtim** [comment]
Many thanx! It worked
